src/org/javarosa/core/model/FormDef.java

```diff
@@ -162,7 +166,7 @@ public static IDataReference getAbsRef(IDataReference ref, TreeReference parentR
     private HashMap<String, DataInstance> formInstances;
     private FormInstance mainInstance = null;
 
-    private HashMap<String, List<Action>> eventListeners;
+    private ActionController actionController;
```

Comment:
How do you feel about moving the initializer here?
`private ActionController actionController = new ActionController();`
Too bad it can’t be final (because of `readExternal`).